packages/shared/src/hooks/post/useViewPost.ts

```diff
@@ -41,17 +40,8 @@ export const useViewPost = (
       const reading = client.getQueryData<ReadingDay[]>(readKey);
 
       if (reading) {
-        const timezoned = getTodayTz(user?.timezone || 'UTC');
-        const date = timezoned.toISOString().split('T')[0];
-        const index = reading.findIndex((read) => read.date === date);
-
-        if (index === -1) {
-          return client.setQueryData(readKey, [...reading, { date, reads: 1 }]);
-        }
-
-        const updatedReading = [...reading];
-        updatedReading[index].reads += 1;
-        return client.setQueryData(readKey, updatedReading);
+        // just mark the query as stale
+        await client.invalidateQueries(readKey);
```

Comment:
I like this approach 👍

Reply:
Generally we always try to do optimistic manipulation historically to avoid extra query invocation when we know the result already.
Just a note on why we do manual cache setting, not against this 👍

Reply:
In this case, we only make the query if data is stale and when the reading streak dropdown is open. I think it should be fine. If we don't open the reading streak popup in 5 minutes, the data will be marked as stale anyway. Which I think would do the optimistic manipulation needlessly in most cases.

Because the logic was so convoluted, it was actually returning the wrong data :) I'd rather have a single point of truth in this case - the backend.

Reply:
Yeah makes total sense indeed :)